File: binary_search.py
def binary_search(ls, num):
    lower = 0
    upper = len(ls)
    while lower < upper:
        mid = (lower + upper) // 2
        if num == ls[mid]:
            return mid
        elif num > ls[mid]:
            if lower == mid: ## in the case where you search for A in [A, B]
                return mid+1
            lower = mid
        elif num < ls[mid]:
            upper = mid
            
## would return len(ls) is num not in ls and more than the largest number or None if smaller than the smallest

File: binarySearch.java
private static void binarySearch(int[] numbers, int searchNum) {
    int[] sortedNumbers = ArraysProblem4.bubbleSort(numbers);
    int lower = 0;
    int upper = numbers.length-1;

    while (lower <= upper) {
        int mid = (lower + upper) / 2;
        if (searchNum < sortedNumbers[mid]) {
            upper = mid - 1;
        } else if (searchNum > sortedNumbers[mid]) {
            lower = mid + 1;
        } else {
            System.out.println(mid);
            return;
        }
    }

    System.out.println("No such number.");
}
